### ISSUE TITLE
typeUserAttrs does not take effect when the field is first added

### ISSUE THREAD

**Issue by baozhoutao** [kevinchappell/formBuilder#915]
### Description:
In the Usage of https://formbuilder.online/docs/formBuilder/options/typeUserAttrs/, add the date type field. The first added date type field does not have the min,max attribute.
### Environment Details:

 - formBuilder Version: 3.1.3
 - Browser: chrome 71.xxx
 - OS:  win10

**Comment by Mordef** [comment]
Got similar:
`const typeUserAttrs = {
    text: {
        dsgvo: {
            label: 'DSGVO',
            value: false,
        }
    }
};`
When insert a textfield i got a textfield, not a checkbox.
On load JSON first, with "dgsvo":true its displayed with checkbox

**Comment by rcbeton** [comment]
I have the same issu.

**Comment by rcbeton** [comment]
I fund the solution for this. You need to set the value attr to get i to get it show on first add.

**Comment by AndriyYushko** [comment]
I also have this problem

**Comment by MarcusDelvecchio** [comment]
value = "" did it for me, wish I saw this before I had the problem though.

**Comment by lucasnetau** [comment]
There are two issues here:

@baozhoutao 

Per the doco 

> Attribute input types are generated using the typeof for the value.

So if there is no 'value' set for the userAttr then userAttrType() returns 'undefined' and the field isn't added.

One to issue here:
* Demo needs to be fixed because this demo doesn't work
* userAttrType() suggests that it should return 'string' as the type if value is undefined per `find(typeCondition => typeCondition[1](attrData))[0] || 'string'` however 'undefined' is a string from typeof so it's never false or null

@Mordef 
> Got similar: `const typeUserAttrs = { text: { dsgvo: { label: 'DSGVO', value: false, } } };` When insert a textfield i got a textfield, not a checkbox. On load JSON first, with "dgsvo":true its displayed with checkbox

This is due to a conversion of the value from _false_ to an empty string.

I can likely fix this with the following change
```javascript
const origValue = tUA.value || (attrValType === 'boolean' ? false : '')
tUA.value = values[attribute] || tUA.value || (attrValType === 'boolean' ? false : '')
```

However the documented way is to assign the value type: 'checkbox' to your config `const typeUserAttrs = { text: { dsgvo: { label: 'DSGVO', value: false, type: 'checkbox',} } };

**Comment by lucasnetau** [comment]
PR #1413 raised for @Mordef's issue

**Comment by lucasnetau** [comment]
PR #1414 raised for original issue raised by @baozhoutao

**Comment by lucasnetau** [comment]
Closing this Issue. Both identified issues have been fixed with recent PRs #1413 and #1414
